### PR TITLE
#342 fix weird behavior of merge when array is encountered

### PR DESCRIPTION
**Pull request by rbarilani**
When merging arrays, the last wins (no special behavior), see #342

### CODE REVIEW

File: server/handler/HookHandler.js [hunk 1]
```diff
@@ -50,7 +51,10 @@ class HookHandler {
         await this.githubService.readZapprFile(owner.login, name, repo.checks[0].token)
         : {}
 
-      const config = merge({}, DEFAULT_CONFIG, zapprUserConfig)
+      const config = mergeWith({}, DEFAULT_CONFIG, zapprUserConfig, (objValue, srcValue) => {
+          // if source value is an array, take that as a value (dont't merge arrays)
+          if(isArray(srcValue)) { return srcValue; }
```
Comment (prayerslayer) [inline server/handler/HookHandler.js]: An array in the user config will overwrite a non-array in the default config?

File: server/handler/HookHandler.js [hunk 2]
```diff
@@ -50,7 +51,10 @@ class HookHandler {
         await this.githubService.readZapprFile(owner.login, name, repo.checks[0].token)
         : {}
 
-      const config = merge({}, DEFAULT_CONFIG, zapprUserConfig)
+      const config = mergeWith({}, DEFAULT_CONFIG, zapprUserConfig, (objValue, srcValue) => {
```
Comment (prayerslayer) [inline server/handler/HookHandler.js]: We can shorten this to

~~~ javascript
const config = mergeWith({}, DEFAULT_CONFIG, zapprUserConfig, (objValue, srcValue) => isArray(srcValue) ? srcValue : undefined)
~~~

**Comment by prayerslayer** [comment]
Thanks, please see my comments 😃

File: server/handler/HookHandler.js (hunk 1 above)
Comment (rbarilani) [inline server/handler/HookHandler.js]: Yes like this yes...

Comment (rbarilani) [inline server/handler/HookHandler.js]: Maybe we can do something like:

```js
const config = mergeWith({}, DEFAULT_CONFIG, zapprUserConfig, (objValue, srcValue) => {
  // if source value is an array, take that as a value (dont't merge arrays)
  if(isArray(srcValue) && (isArray(objValue) || typeof objValue === 'undefined' )) {
    return srcValue;
  }
});
```

@prayerslayer WDYT?

Comment (prayerslayer) [inline server/handler/HookHandler.js]: Yup, better :)

**Comment by prayerslayer** [comment]
👍

**Comment by rbarilani** [comment]
@prayerslayer Ok updated. considering your comment. The shorten version has discarded because we apply the special array behavior only if the objValue it's an array or is undefined.

**Comment by prayerslayer** [comment]
Closing this to save you some work. Follow #350 😃
